src/OrleansProviders/Streams/Common/Monitors/IObjectPoolMonitor.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Monitor track block pool related metrics
+    /// </summary>
+    public interface IObjectPoolMonitor
+    {
+        #region Event driven metrics
+        /// <summary>
+        /// Track every time cache allocate memory from block pool
+        /// </summary>
+        /// <param name="memoryInByte"></param>
+        void TrackMemroyAllocatedByCache(int memoryInByte);
```

Comment:
Also I think we should report this by object count, not memory size.  Object pools rarely know the size of what their pooling.
If this is specifically for a block pool, we can include memory size, but if this is for object pools in general, size probably won't work.